### DISCUSSION TITLE
Add mouse event to Marker

### DISCUSSION THREAD

**Discussion by AFA141** [mapsforge/mapsforge#1323] [Q&A] [answered]
What is best way to add mouse event when i right click on Marker?

**Comment by devemux86** [comment]
If you mean the Desktop, there is an implementation in [MouseEventListener](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-awt/src/main/java/org/mapsforge/map/awt/input/MouseEventListener.java).

You can customize it according to your app needs
and override [MapView.addListeners](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-awt/src/main/java/org/mapsforge/map/awt/view/MapView.java#L106) to use it.

**Reply by AFA141** [reply]
Yes i mean the Desktop, i saw that class
How i can know if user clicked on Marker?

**Comment by devemux86** [comment, marked as answer]
> How i can know if user clicked on Marker?

It is demonstrated in [MouseEventListener.mouseClicked](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-awt/src/main/java/org/mapsforge/map/awt/input/MouseEventListener.java#L40).

And need tappable markers, like in [Utils.createTappableMarker](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/Utils.java#L85).

**Reply by AFA141** [reply]
Thanks
